src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/OmniSharpStrongNamedExports.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Composition;
+using Microsoft.AspNetCore.Razor.OmniSharpPlugin.StrongNamed;
+using OmniSharp;
+
+namespace Microsoft.AspNetCore.Razor.OmniSharpPlugin
+{
+    // We need to re-export MEF based services from the OmniSharp plugin strong named assembly in order
+    // to make those services available via MEF. This isn't an issue for Roslyn based services because
+    // we're able to hook into OmniSharp's Roslyn service aggregator to allow it to inspect the strong
+    // named plugin assembly.
```

Comment:
How long term is all this going to be? Is there a plan to remove all this hackery in the future?

Reply:
No plan to remove unless OmniSharp decides to strong name their assemblies or we decide to make our project system public. I'd bet on the latter